File: jokenpo/respostas/jokenpo.py
# -*- coding: utf-8 -*-
import random
import os

def jogada_pc():
    jogada = int(random.uniform(1, 4))
    return jogada

def pegar_nome(jogada):
    if jogada == 1:
        return "Pedra"
    if jogada == 2:
        return "Papel"
    return "Tesoura"

def checkGame(jogador, pc):
    if jogador == 1 and pc == 2:
        return 2
    elif jogador == 1 and pc == 3:
		return 1
    elif jogador == 2 and pc == 1:
		return 1
    elif jogador == 2 and pc == 3:
		return 2
    elif jogador == 3 and pc == 1:
		return 2
    elif jogador == 3 and pc == 2:
		return 1
    return 0

if __name__ == '__main__':
    round = 1
    pc = 0
    jogador = 0
    vitoriasPC = 0
    vitoriasJogador = 0
    continuar = 0

    while(round <= 5):
        print "\n# JOKENPO #\n"
        print "Você: " + str(vitoriasJogador) + " | PC: " + str(vitoriasPC) + "\n"
        print "Round: " + str(round) + " / 5\n"
        print "1 - Pedra\n2 - Papel\n3 - Tesoura\n"

        jogador = input(">> ")
        pc = jogada_pc();

        print "\nVocê jogou: " + pegar_nome(jogador) + " e o PC: " + pegar_nome(pc)

        if checkGame(jogador, pc) == 1:
            print "Você ganhou!\n"
            vitoriasJogador += 1
        elif checkGame(jogador, pc) == 2:
            print "O PC ganhou!\n"
            vitoriasPC += 1
        else:
            print "Ninguém ganhou!\n"

        if round == 5:
            print "\nPLACAR FINAL:\n"
            print "Você: " + str(vitoriasJogador) + " | PC: " + str(vitoriasPC)
            print "\nJOGAR NOVAMENTE?\n1 - SIM | 0 - NÃO\n"

            continuar = input(">> ")

            if continuar == 1:
                os.system("clear");
                round = 0
                pc = 0
                jogador = 0
                vitoriasPC = 0
                vitoriasJogador = 0
                continuar = 0
        round += 1

File: jokenpo/respostas/jokenpo.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int jogada_pc(){
	int jogada = (rand() % 3) + 1;
  return jogada;
}

const char * pegar_nome(int jogada){
  if(jogada == 1)
    return "Pedra";
  if(jogada == 2)
    return "Papel";
  return "Tesoura";
}

int checkGame(int jogador, int pc) {
	if(jogador == 1 && pc == 2)
		return 2;
	if(jogador == 1 && pc == 3)
		return 1;

	if(jogador == 2 && pc == 1)
		return 1;
	if(jogador == 2 && pc == 3)
		return 2;

	if(jogador == 3 && pc == 1)
		return 2;
	if(jogador == 3 && pc == 2)
		return 1;

  return 0;
}

int main() {
  srand(time(NULL));
  int round = 1, pc = 0, jogador = 0, vitoriasPC = 0, vitoriasJogador = 0, continuar = 0;
  while(round <= 5){
    printf("\n# JOKENPO #\n");
		printf("Você: %d | PC: %d\n", vitoriasJogador, vitoriasPC);
		printf("Round: %d / 5\n", round);
		printf("1 - Pedra\n2 - Papel\n3 - Tesoura\n");

    printf(">> ");
    scanf("%d", &jogador);
    pc = jogada_pc();

    printf("\nVocê jogou: %s e o PC: %s\n", pegar_nome(jogador), pegar_nome(pc));

    if(checkGame(jogador, pc) == 1){
      printf("Você ganhou!\n");
      vitoriasJogador++;
    }else if(checkGame(jogador, pc) == 2){
      printf("O PC ganhou!\n");
      vitoriasPC++;
    }else{
      printf("Ninguém ganhou!\n");
    }

    if(round == 5){
      printf("\nPLACAR FINAL:\n");
			printf("Você: %d | PC: %d\n", vitoriasJogador, vitoriasPC);
			printf("\nJOGAR NOVAMENTE?\n1 - SIM | 0 - NÃO\n");
			printf(">> ");
			scanf("%d", &continuar);
      if(continuar == 1){
				system("clear");
        round = 0, pc = 0, jogador = 0, vitoriasPC = 0, vitoriasJogador = 0, continuar = 0;
      }
    }
    round++;
  }
	return 0;
}
